eis_toolkit/raster_processing/unique_combinations.py

```diff
@@ -0,0 +1,68 @@
+from typing import List, Dict
+
+import itertools
+import numpy as np
+import rasterio
+
+from eis_toolkit.exceptions import InvalidParameterValueException
+
+
+def _unique_combinations(  # type: ignore[no-any-unimported]
+    list_of_rasters: List[rasterio.io.DatasetReader],
+    path_to_output_file: str
+) -> rasterio.io.DatasetReader:
+
+    bands = []
+    for raster in list_of_rasters:
+        bands.append(raster.read())
+
+    output_band = np.empty_like(bands[0])
+    combinations: Dict[tuple, int] = {}
+    for row, column in itertools.product(range(len(bands[0])), range(len(bands[0][0]))):
+
+        combination = tuple(bands[band][row][column] for band, _ in enumerate(bands))
+
```

Comment:
At the moment this does not work, but after the bands array will actually have bands as elements it should work I think.